docs/rules/no-unused-expressions.md

```diff
@@ -46,7 +58,11 @@ The following patterns are not considered problems by default:
 ```js
 /*eslint no-unused-expressions: 2*/
 
-{}
+{} // In this context, this is a block statement, not an object literal
+
+{myLabel: anExpression()} // In this context, this is a block statement with a label and expression, not an object literal
+
+function aFunctionDeclaration () {}
```

Comment:
This might open the door to objection that a declaration is not an expression, therefore the rule does not apply.